install_script_opw.sh

```diff
@@ -0,0 +1,644 @@
+    echo "$ERROR_MESSAGE"
+    report_telemetry
+    exit 1;
+  fi
+  worker_dist_channel=$DD_WORKER_DIST_CHANNEL
+fi
+
+if [ -n "$TESTING_YUM_VERSION_PATH" ]; then
+  yum_version_path=$TESTING_YUM_VERSION_PATH
+else
+  yum_version_path="${worker_dist_channel}/observability-pipelines-worker-1"
+fi
+
+if [ -n "$TESTING_APT_REPO_VERSION" ]; then
+  apt_repo_version=$TESTING_APT_REPO_VERSION
+else
+  apt_repo_version="${worker_dist_channel} observability-pipelines-worker-1"
+fi
+
+if [ ! "$apikey" ]; then
+  if [ ! -e "$config_file" ]; then
+    printf "\033[31mAPI key not available in DD_API_KEY environment variable.\033[0m\n"
+    exit 1;
+  fi
+fi
+
+# OS/Distro Detection
+# Try lsb_release, fallback with /etc/issue then uname command
+KNOWN_DISTRIBUTION="(Debian|Ubuntu|RedHat|CentOS|Amazon)"
+DISTRIBUTION=$(lsb_release -d 2>/dev/null | grep -Eo $KNOWN_DISTRIBUTION  || grep -Eo $KNOWN_DISTRIBUTION /etc/issue 2>/dev/null || grep -m1 -Eo $KNOWN_DISTRIBUTION /etc/os-release 2>/dev/null || uname -s)
+
+if [ "$DISTRIBUTION" = "Darwin" ]; then
+    ERROR_MESSAGE="This script does not support installing on Mac."
+    ERROR_CODE=$UNSUPPORTED_PLATFORM_CODE
+    printf "\033[31m$ERROR_MESSAGE
+\033[0m\n"
+    report_telemetry
+    exit 1;
+
+elif [ -f /etc/debian_version ] || [ "$DISTRIBUTION" == "Debian" ] || [ "$DISTRIBUTION" == "Ubuntu" ]; then
+    OS="Debian"
+elif [ -f /etc/redhat-release ] || [ "$DISTRIBUTION" == "RedHat" ] || [ "$DISTRIBUTION" == "CentOS" ] || [ "$DISTRIBUTION" == "Amazon" ]; then
+    OS="RedHat"
+# Some newer distros like Amazon may not have a redhat-release file
+elif [ -f /etc/system-release ] || [ "$DISTRIBUTION" == "Amazon" ]; then
+    OS="RedHat"
+fi
+
+# Root user detection
+if [ "$(echo "$UID")" = "0" ]; then
+    sudo_cmd=''
+else
+    sudo_cmd='sudo'
+fi
+
+# Install the necessary package sources
+if [ "$OS" = "RedHat" ]; then
+    echo -e "\033[34m\n* Installing YUM sources for Datadog\n\033[0m"
+
+    UNAME_M=$(uname -m)
+    if [ "$UNAME_M" == "aarch64" ]; then
+        ARCHI="aarch64"
+    else
+        ARCHI="x86_64"
+    fi
+
+    # Because of https://bugzilla.redhat.com/show_bug.cgi?id=1792506, we disable
+    # repo_gpgcheck on RHEL/CentOS 8.1
+    if [ -z "$rpm_repo_gpgcheck" ]; then
+        if grep -q "8\.1\(\b\|\.\)" /etc/redhat-release 2>/dev/null; then
+            rpm_repo_gpgcheck=0
+        else
+            rpm_repo_gpgcheck=1
+        fi
+    fi
+
+    gpgkeys="https://${keys_url}/DATADOG_RPM_KEY.public\n       https://${keys_url}/DATADOG_RPM_KEY_E09422B3.public"
+
+    gpgkeys=''
+    separator='\n       '
+    for key_path in "${RPM_GPG_KEYS[@]}"; do
+      gpgkeys="${gpgkeys:+"${gpgkeys}${separator}"}https://${keys_url}/${key_path}"
+    done
+
+    $sudo_cmd sh -c "echo -e '[datadog]\nname = Datadog, Inc.\nbaseurl = https://${yum_url}/${yum_version_path}/${ARCHI}/\nenabled=1\ngpgcheck=1\nrepo_gpgcheck=${rpm_repo_gpgcheck}\npriority=1\ngpgkey=${gpgkeys}' > /etc/yum.repos.d/datadog.repo"
```

Comment:
The filename has to be something else than `datadog.repo`, e.g. `datadog-observability-pipelines-worker.repo` that the public OPW instructions propose. Same goes for the `[datadog]` repo ID - let's use what we have in https://docs.datadoghq.com/observability_pipelines/setup/datadog/?tab=rpmbasedlinux to be consistent.

Reply:
Completely overlooked that, good catch!

Reply:
Resolved in [5bda0aa](https://github.com/DataDog/agent-linux-install-script/pull/63/commits/5bda0aa8b2b5d7b4c3262460d7012aff26b1a4c5)